src/unitxt/splitters.py

```diff
@@ -151,13 +153,52 @@ def filter_source_by_instance(
 
 
 class RandomSampler(Sampler):
+    """Selects a random sample of instances."""
+
     def sample(
-        self, instances_pool: List[Dict[str, object]]
+        self,
+        instances_pool: List[Dict[str, object]],
+        instance: Optional[Dict[str, object]] = None,
     ) -> List[Dict[str, object]]:
         instances_pool = list(instances_pool)
         return self.random_generator.sample(instances_pool, self.sample_size)
 
 
+class CloseTextSampler(Sampler):
+    """Selects the samples of instances which are the closest textual match to the given instance.
+
+    Comparison is done based on a given field in the instance.
+
+    """
+
+    field: str
```

Comment:
Maybe have a None default that uses just the serialization of the inputs?

Reply:
This seems bit of an odd default.  Consider inputs such as "text_type" or "classes" , which you don't want to take part in the the calculation.

Reply:
The do not harm either. Anyways just a suggestion